src/Evolution/EventsAndDenseTriggers/EventsAndDenseTriggers.hpp

```diff
@@ -251,6 +251,11 @@ void EventsAndDenseTriggers::run_events(
     for (const auto& event : trigger->events) {
       event->run(box, cache, array_index, component);
     }
+    db::mutate<::evolution::Tags::PreviousTriggerTime>(
+        make_not_null(&box), [](const gsl::not_null<std::optional<double>*>
+                                    previous_trigger_time) noexcept {
+          *previous_trigger_time = std::numeric_limits<double>::signaling_NaN();
```

Comment:
what is the difference between setting `previous_trigger_time` to signaling_NaN vs `std::null_opt`?

Reply:
An empty optional means this is the first time the event has run.  A NaN (new in this PR) means you're not even in an event so asking for the previous time doesn't make sense.

Reply:
Okay, thanks. Could you add a comment to the tag (or wherever you feel is most appropriate)?

Reply:
It already has one.  This just makes the code crash if you do it wrong instead of continuing with a nonsense value.

```
/*!                                                                             
 * \ingroup EventsAndTriggersGroup                                              
 * \brief Previous time at which the trigger activated.                         
 *                                                                              
 * \details This tag is populated with the most recent time the current trigger 
 * fired prior to the current activation.                                       
 * \note This tag is only populated with a meaningful value for events invoked  
 * from `EventsAndDenseTriggers`; during other actions, the tag should not be   
 * used.                                                                        
 */
```

Reply:
sorry, so it does; I should read the documentation more carefully...